tests/NuGet.IndexingTests/PackageAnalyzerTests.cs

```diff
@@ -0,0 +1,125 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using NuGet.Indexing;
+using NuGet.IndexingTests.TestSupport;
+using Xunit;
+
+namespace NuGet.IndexingTests
+{
+    public class PackageAnalyzerTests
+    {
+        [Theory, MemberData("TheoryData")]
+        public void AddsCorrectFieldAnalyzers(string field, string text, TokenAttributes[] expected)
+        {
+            // ARRANGE
+            var analyzer = new PackageAnalyzer();
+
+            // ACT
+            var tokenStream = analyzer.TokenStream(field, new StringReader(text));
+            var tokenAttributes = tokenStream.GetTokenAttributes().ToArray();
+
+            // ASSERT
+            Assert.Equal(expected, tokenAttributes);
+        }
+
+        public static IEnumerable<object[]> TheoryData
+        {
+            get
+            {
+                yield return new object[]
+                {
+                    "Id",
+                    "AaBbCc",
```

Comment:
I tried to limit the test cases in PackageAnalyzer to only verify that the correct analyzer per field is selected. I added new test classes for each analyzer to cover these additional, deeper cases.